### PR TITLE
Feature: Upgrade to Java 17 (Azul Zulu)

### PR DESCRIPTION
**Pull request by joschi**
- [x] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

Users of Kafka UI will have to provide a Java 17 (or higher) runtime environment in order to run the application, if not using the Docker image.

**What changes did you make?** (Give an overview)

Java/OpenJDK 13 has reached end of life (and end of security support) more than 2.5 years ago: https://endoflife.date/java

This change set updates the build, CI pipeline, and Docker image of Kafka UI to Java 17 (LTS).

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [x] Unit checks
- [x] Integration checks
- [x] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**
A cute wombat

### CODE REVIEW

**Comment by joschi** [comment]
GitHub Actions still seems to pick up the old workflow files: https://github.com/provectus/kafka-ui/actions/runs/3392395053/jobs/5638552698#step:4:3

But all the occurrences of `actions/setup-java@v1` has been replaced by `actions/setup-java@v3`. 🤔

**Comment by joschi** [comment]
Example run, failing due to missing secrets for SonarQube and Qase but building the Maven project and running unit tests worked fine: https://github.com/personio/kafka-ui/pull/13

File: kafka-ui-e2e-checks/pom.xml [hunk 1]
```diff
@@ -267,19 +267,10 @@
                     <plugin>
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-surefire-plugin</artifactId>
-                        <version>${maven.surefire-plugin.version}</version>
                         <configuration>
                             <skipTests>true</skipTests>
                         </configuration>
                     </plugin>
-                    <plugin>
```
Comment (Haarolean) [inline kafka-ui-e2e-checks/pom.xml:275]: where did this plugin go? for local profile

**Review by Haarolean — CHANGES_REQUESTED** [review]
Hey, thanks for the contribution.
A few questions:
1) why did you decide to change the distro (openjdk) to zulu?
2) why have you removed cache@v3 action everywhere?

**Comment by Haarolean** [comment]
> GitHub Actions still seems to pick up the old workflow files: https://github.com/provectus/kafka-ui/actions/runs/3392395053/jobs/5638552698#step:4:3
> 
> But all the occurrences of `actions/setup-java@v1` has been replaced by `actions/setup-java@v3`. 🤔

That's fine. It's a github security thing, you can't run the worflow you've changed in a branch.

File: kafka-ui-e2e-checks/pom.xml (hunk 1 above)
Comment (joschi) [reply kafka-ui-e2e-checks/pom.xml:275]: The Maven compiler plugin is available by default and the version and the Java release are defined in the parent POM of the `kafka-ui-e2e-checks` project:


https://github.com/provectus/kafka-ui/blob/1f65a905b8e23d493114681c7b5772d6a8a65d51/pom.xml#L14

https://github.com/provectus/kafka-ui/blob/1f65a905b8e23d493114681c7b5772d6a8a65d51/pom.xml#L92-L96

This being said, happy to add the plugin back for clarity.

**Comment by joschi** [comment]
> 1. why did you decide to change the distro (openjdk) to zulu?

@Haarolean That's the same distribution of OpenJDK which is being used in all the GitHub Actions workflows. Additionally, it offers Docker base images based on Alpine Linux for `amd64` and `aarch64`.

Previously, the Kafka UI Docker image was using AdoptOpenJDK which is nowadays named Eclipse Temurin (under the Adoptium umbrella) and it doesn't offer an official Alpine Linux base image for `aarch64`.

> 2. why have you removed cache@v3 action everywhere?

This functionality is covered by `actions/setup-java@v3` nowadays and doesn't require a separate caching step via `actions/cache`: https://github.com/actions/setup-java#caching-packages-dependencies

File: kafka-ui-e2e-checks/pom.xml (hunk 1 above)
Comment (Haarolean) [reply kafka-ui-e2e-checks/pom.xml:275]: Nah, that's alright, thank you

**Comment by Haarolean** [comment]
> @Haarolean That's the same distribution of OpenJDK which is being used in all the GitHub Actions workflows. Additionally, it offers Docker base images based on Alpine Linux for `amd64` and `aarch64`.

> Previously, the Kafka UI Docker image was using AdoptOpenJDK which is nowadays named Eclipse Temurin (under the Adoptium umbrella) and it doesn't offer an official Alpine Linux base image for `aarch64`.

Thanks a lot. I've got a bit out of context with jdk updates recently.

> This functionality is covered by `actions/setup-java@v3` nowadays and doesn't require a separate caching step via `actions/cache`: https://github.com/actions/setup-java#caching-packages-dependencies

Alright, thanks.

**Review by Haarolean — APPROVED** [review]

**Pull request merged**